### PR TITLE
Enable IMA test scenario for systems with TPM emulator

### PR DESCRIPTION
**Pull request by kkaarreell**

### CODE REVIEW

File: setup/configure_kernel_ima_module/test.sh [hunk 1]
```diff
@@ -24,9 +25,13 @@ rlJournalStart
...
         # install IMA policy
-        rlRun "limeInstallIMAConfig"
+        rlRun "limeInstallIMAConfig ${IMA_POLICY_FILE}"
         # clear TPM
-        rlRun "tpm2_clear"
+        if ! limeTPMEmulated; then
+            rlRun "tpm2_clear"
+        fi
+        # workaround for issue https://github.com/keylime/keylime/issues/1025
+        rlRun "echo 'd /var/run/keylime 0700 keylime keylime' > /usr/lib/tmpfiles.d/keylime.conf"
```
Comment (sergio-correia) [inline setup/configure_kernel_ima_module/test.sh:34]: Isn't this expected to be in the packaging?

**Review by Koncpa — APPROVED** [review]

File: setup/configure_kernel_ima_module/test.sh (hunk 1 above)
Comment (kkaarreell) [reply setup/configure_kernel_ima_module/test.sh:34]: It is. But without a workaround in place I can't run tests successfully and it takes time to get issue 1025 fixed in distribution packages. I plan to remove it once the fix is in place.

**Comment by kkaarreell** [comment]
CI test results revealed some issue that I need to investigate. One failure in the distribution test plan is due to using keylime 6.4.0. 
There is also some gpg issue with RPM signing. I assume there is some problem with terminal and pinentry. We should probably configure rpm to use password file instead. 
The other failures I do not know yet.

**Comment by kkaarreell** [comment]
> The failure I need to investigate is the second iteration of /functional/basic-attestation-on-localhost
> 
> ```
> 2022-06-10 14:40:42.501 - keylime.tenant - INFO - Agent Info:
> {"d432fbb3-d2f1-4a97-9ef7-75bd81c00000": {"operational_state": "Invalid Quote", "v": "Dh7PO6ejyd1mZmfsQgDFzxqLp6jZcfyVymW+4AFFh7s=", "ip": "127.0.0.1", "port": 9002, "tpm_policy": "{\"22\": [\"0000000000000000000000000000000000000001\", \"0000000000000000000000000000000000000000000000000000000000000001\", \"000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001\", \"ffffffffffffffffffffffffffffffffffffffff\", \"ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff\", \"ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff\"], \"15\": [\"0000000000000000000000000000000000000000\", \"0000000000000000000000000000000000000000000000000000000000000000\", \"000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000\"], \"mask\": \"0x408400\"}", "meta_data": "{\"cert_serial\": 3, \"subject\": \"OU=53,O=MITLL,L=Lexington,ST=MA,CN=d432fbb3-d2f1-4a97-9ef7-75bd81c00000,C=US\"}", "allowlist_len": 6, "mb_refstate_len": 0, "accept_tpm_hash_algs": ["sha512", "sha384", "sha256", "sha1"], "accept_tpm_encryption_algs": ["ecc", "rsa"], "accept_tpm_signing_algs": ["ecschnorr", "rsassa"], "hash_alg": "sha256", "enc_alg": "rsa", "sign_alg": "rsassa", "verifier_id": "default", "verifier_ip": "127.0.0.1", "verifier_port": 8881, "severity_level": 6, "last_event_id": "ima.pcr_mismatch"}}
> :: [ 14:40:43 ] :: [   FAIL   ] :: Command 'limeWaitForAgentStatus d432fbb3-d2f1-4a97-9ef7-75bd81c00000 'Get Quote'' (Expected 0, got 1)
> ```

I believe I have fixed that one. It was a typo of the test, of course.

**Comment by kkaarreell** [comment]
So Fedora failures are gone but RHEL ones are still remaining. I have no idea what is going on with RHEL-9. :-(

**Comment by kkaarreell** [comment]
Failures on RHEL-9 are due to IMA signatures installed by rpm-plugin-ima. It seems that these make attestation to fail despite paths being on the excludelist.

**Comment by kkaarreell** [comment]
So now tests have passed 3 times in a row. It seems they are stable, even though I do not fully understand what was the issue on Fedora.

**Comment by kkaarreell** [comment]
It seems to be passing reliably. I think we are good to go.

**Review by sergio-correia — APPROVED** [review]

**Pull request merged**